src/swiper-vue.d.ts

```diff
@@ -0,0 +1,567 @@
+    slideVisibleClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    slideDuplicateClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    slideNextClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    slideDuplicateNextClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    slidePrevClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    slideDuplicatePrevClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    wrapperClass: {
+      type: StringConstructor;
+      default: undefined;
+    };
+    runCallbacksOnInit: {
+      type: BooleanConstructor;
+      default: undefined;
+    };
+    observer: { type: BooleanConstructor; default: undefined };
+    observeParents: {
+      type: BooleanConstructor;
+      default: undefined;
+    };
+    observeSlideChildren: {
+      type: BooleanConstructor;
+      default: undefined;
+    };
+    a11y: {
+      type: PropType<A11yOptions | boolean>;
+      default: undefined;
+    };
+    autoplay: {
+      type: PropType<AutoplayOptions | boolean>;
+      default: undefined;
+    };
+    controller: {
+      type: PropType<ControllerOptions>;
+      default: undefined;
+    };
+    coverflowEffect: {
+      type: PropType<CoverflowEffectOptions>;
+      default: undefined;
+    };
+    cubeEffect: {
+      type: PropType<CubeEffectOptions>;
+      default: undefined;
+    };
+    fadeEffect: {
+      type: PropType<FadeEffectOptions>;
+      default: undefined;
+    };
+    flipEffect: {
+      type: PropType<FlipEffectOptions>;
+      default: undefined;
+    };
+    hashNavigation: {
+      type: PropType<HashNavigationOptions | boolean>;
+      default: undefined;
+    };
+    history: {
+      type: PropType<HistoryOptions | boolean>;
+      default: undefined;
+    };
+    keyboard: {
+      type: PropType<KeyboardOptions | boolean>;
+      default: undefined;
+    };
+    lazy: {
+      type: PropType<LazyOptions | boolean>;
+      default: undefined;
+    };
+    mousewheel: {
+      type: PropType<MousewheelOptions | boolean>;
+      default: undefined;
+    };
+    navigation: {
+      type: PropType<NavigationOptions | boolean>;
+      default: undefined;
+    };
+    pagination: {
+      type: PropType<PaginationOptions | boolean>;
+      default: undefined;
+    };
+    parallax: {
+      type: PropType<ParallaxOptions | boolean>;
+      default: undefined;
+    };
+    scrollbar: {
+      type: PropType<ScrollbarOptions | boolean>;
+      default: undefined;
+    };
+    thumbs: { type: PropType<ThumbsOptions>; default: undefined };
+    virtual: {
+      type: PropType<VirtualOptions | boolean>;
+      default: undefined;
+    };
+    zoom: {
+      type: PropType<ZoomOptions | boolean>;
+      default: undefined;
+    };
+  },
+  () => JSX.Element,
+  unknown,
+  {},
+  {},
+  ComponentOptionsMixin,
+  ComponentOptionsMixin,
+  {
```

Comment:
I think events here are in totally same format as `SwiperEvents` in `./types/swiper-events.d.ts`, maybe we can reuse it here?

Reply:
I can't seem to make it work with `SwiperEvents`, as it's complaining about it not respecting a constraint for `EmitsOptions`.

Using something like `[event in keyof SwiperEvents]?: SwiperEvents[event]` gets rid of the errors, but then all the events disappear from auto-completions provided by Vetur or Volar.

They are pretty much the same though, so it'd be good to figure out a way to use them here 😕